### PR TITLE
[rec, stat] Added user based recommendation, set up broad genre comparison

### PR DESCRIPTION
**Pull request by hannahgz**
- Set up cache to hold top "k" songs from each category that updates async while updating database.
- Added new columns to Rec table that represent cumulative sum. 
- Created score which equals ((user_danceability * song_danceability + user_acousticness * song_acousticness)) b/c everything normalized between 0-1.

Ensure Rec table has cumulative sum
image
Each Song has attributes added
image
New post submission still performs as expected
image

### CODE REVIEW

**Comment by hannahgz** [comment]
Additional Test Plan:

Genre comparison behaves as expected
image

Adding new songs load in user-based rec + no duplicates
image

File: project_start/express-api/routes/post.js [hunk 1]
```diff
@@ -118,6 +139,45 @@ router.post("/new-post", async (req, res, next) => {
   }
 });
 
+//https://stackoverflow.com/questions/25500316/sort-a-dictionary-by-value-in-javascript
+router.post("/new-post-rec", async (req, res, next) => {
+  try {
+    const { audioFeatures } = req.body;
+    // Add to Rec table
+    const Rec = Parse.Object.extend("Rec");
```
Comment (mohan-upadhyay) [inline project_start/express-api/routes/post.js:147]: You can use better naming conventions than Rec. Table name should be descriptive of the entity/relationship it is representing

File: project_start/express-api/routes/post.js [hunk 2]
```diff
@@ -4,19 +4,16 @@ var request = require("request");
...
 const { default: axios } = require("axios");
-// Parse.initialize(
-//   "z81Jsr6Tc1lcHyxZK7a5psWRFOBuOs2e0nxXudMj",
-//   "JTrwOsEpJabYLzZVqKuG07FD5Lxwm2SzhM5EUVt5"
-// );
 
 Parse.initialize(
-  "YmpmeHLzpiEt1IiupexyPzd9vCgETDvaeW2rWh0U",
-  "8xOR0nDLQMijBppInKvJFsLXcDDfl7RwQ1d2QnNS"
+  "8mJaCOPGxTw5RVUHZ8Dfqx8oaZ5H0N4gTtfeIkrE",
```
Comment (mohan-upadhyay) [inline project_start/express-api/routes/post.js:9]: What are these keys?

File: project_start/express-api/routes/profile.js [hunk 3]
```diff
@@ -3,10 +3,10 @@ const router = express.Router();
 var request = require("request");
 
 const Parse = require("parse/node");
-// Parse.initialize("z81Jsr6Tc1lcHyxZK7a5psWRFOBuOs2e0nxXudMj", "JTrwOsEpJabYLzZVqKuG07FD5Lxwm2SzhM5EUVt5")
+
 Parse.initialize(
-  "YmpmeHLzpiEt1IiupexyPzd9vCgETDvaeW2rWh0U",
-  "8xOR0nDLQMijBppInKvJFsLXcDDfl7RwQ1d2QnNS"
+  "8mJaCOPGxTw5RVUHZ8Dfqx8oaZ5H0N4gTtfeIkrE",
```
Comment (mohan-upadhyay) [inline project_start/express-api/routes/profile.js:8]: Seems like these keys are used at multiple places. Would be more efficient to store it in a common file.

File: project_start/express-api/routes/profile.js [hunk 4]
```diff
@@ -3,10 +3,10 @@ const router = express.Router();
 var request = require("request");
 
 const Parse = require("parse/node");
-// Parse.initialize("z81Jsr6Tc1lcHyxZK7a5psWRFOBuOs2e0nxXudMj", "JTrwOsEpJabYLzZVqKuG07FD5Lxwm2SzhM5EUVt5")
```
Comment (mohan-upadhyay) [inline project_start/express-api/routes/profile.js:6]: remove comment

File: project_start/express-api/routes/post.js (hunk 1 above)
Comment (hannahgz) [reply project_start/express-api/routes/post.js:147]: I fixed the naming convention in the pull request after this one!

File: project_start/express-api/routes/post.js (hunk 2 above)
Comment (hannahgz) [reply project_start/express-api/routes/post.js:9]: These are the "Application ID" and "JavaScript key" (both API keys from Parse).

File: project_start/express-api/routes/profile.js (hunk 3 above)
Comment (hannahgz) [reply project_start/express-api/routes/profile.js:8]: Will do!

File: project_start/express-api/routes/profile.js (hunk 4 above)
Comment (hannahgz) [reply project_start/express-api/routes/profile.js:6]: Got it!

**Pull request merged**